src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -60,28 +84,125 @@ public class NewAzureVMCommand : VirtualMachineBaseCmdlet
         public PSVirtualMachine VM { get; set; }
 
         [Parameter(
-           Mandatory = false,
-           Position = 3,
-           ValueFromPipelineByPropertyName = true)]
+            ParameterSetName = DefaultParameterSet,
+            Mandatory = false,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true)]
         [ValidateNotNullOrEmpty]
         public string[] Zone { get; set; }
 
         [Parameter(
+            ParameterSetName = DefaultParameterSet,
+            Position = 3,
             HelpMessage = "Disable BG Info Extension")]
         public SwitchParameter DisableBginfoExtension { get; set; }
 
         [Parameter(
+            ParameterSetName = DefaultParameterSet,
             Mandatory = false,
             ValueFromPipelineByPropertyName = true)]
         public Hashtable Tags { get; set; }
 
         [Parameter(
+            ParameterSetName = DefaultParameterSet,
             Mandatory = false,
             ValueFromPipelineByPropertyName = false)]
         [ValidateNotNullOrEmpty]
         public string LicenseType { get; set; }
 
+        [Parameter(
+            ParameterSetName = StrategyParameterSet,
+            Mandatory = true)]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = StrategyParameterSet,
+            Mandatory = false)]
+        public string AddressPrefix { get; set; } = "192.168.0.0/16";
+
+        [Parameter(
+            ParameterSetName = StrategyParameterSet,
+            Mandatory = false)]
+        public string SubnetAddressPrefix { get; set; } = "192.168.1.0/24";
+
+        [Parameter(ParameterSetName = StrategyParameterSet, Mandatory = true)]
+        public PSCredential Credential { get; set; }
+
         public override void ExecuteCmdlet()
+        {
+            switch (ParameterSetName)
+            {
+                case StrategyParameterSet:
+                    StrategyExecuteCmdlet();
+                    break;
+                default:
+                    DefaultExecuteCmdlet();
+                    break;
+            }
+        }
+
+        private sealed class Client : IClient
+        {
+            public string SubscriptionId { get; }
+
+            IAzureContext Context { get; }
+
+            public Client(IAzureContext context)
+            {
+                Context = context;
+                SubscriptionId = Context.Subscription.Id;
+            }
+
+            public T GetClient<T>()
+                where T : ServiceClient<T>
+                => AzureSession.Instance.ClientFactory.CreateArmClient<T>(
+                    Context, AzureEnvironment.Endpoint.ResourceManager);
+        }
+
+        public void StrategyExecuteCmdlet()
+        {
+            if (ResourceGroupName == null)
+            {
+                ResourceGroupName = Name;
+            }
+
+            var resourceGroup = ResourceGroupStrategy.CreateResourceGroupConfig(ResourceGroupName);
```

Comment:
@sergey-shandar we should wrap the part of this code that is creating the objects with the `ShouldProcess` block and let them know the name of the VM they are creating and what resource group all necessary resources will be created in.

Reply:
Done.